test/src/dataEntities/appointment.test.ts

```diff
@@ -189,7 +204,19 @@ describe("Appointment", () => {
         const testAppointment = Appointment.parse(clone);
         const signedAppointment = await sign(testAppointment, customerSigner);
 
-        await signedAppointment.validate();
+        await signedAppointment.validate(blockCache, pisaContractAddress);
+    });
+
+    // TODO:274:should not be able to specufy "uint8[]", [[2]] - with a 2 in here
+    fnIt<Appointment>(a => a.validate, "can specify none of the indexed arguments", async () => {
```

Comment:
TODO

Reply:
It turns out there's no easy way to do this - the only way I can think of is to parse the arguments one by one, substringing the rest as we go. I'll make a new ticket for this, but I'm actually considering changing this encoding method, perhaps it would be better to just encode the topic client side. This would give us less control over what events we could watch, but I dont think that would matter. #293